src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentContext.cs

```diff
@@ -158,11 +158,7 @@ async ValueTask<SourceText> GetHtmlSourceTextCoreAsync(CancellationToken cancell
 
         static RazorSyntaxNode? GetSyntaxNodeCore(RazorCodeDocument codeDocument, int absoluteIndex)
         {
-            var syntaxTree = codeDocument.GetSyntaxTree().AssumeNotNull();
-
-            return syntaxTree.Root is RazorSyntaxNode root
-                ? root.FindInnermostNode(absoluteIndex)
-                : null;
+            return codeDocument.GetRequiredSyntaxRoot().FindInnermostNode(absoluteIndex);
```

Comment:
>GetRequiredSyntaxRoot [](http://example.com/codeflow?start=32&length=21)

nit: used to not throw if tree.Root was null.

Reply:
Yes, but this was wrong and unnecessarily paranoid, so I changed it here. `RazorSyntaxTree.Root` is annotated as non-null and `RazorSyntaxTree's` constructor throws if `root` is null: https://github.com/dotnet/razor/blob/a81a6da8372912daa2bada10dbe319569b245a45/src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/RazorSyntaxTree.cs#L20-L34